Yoast/wordpress-seo
# 14386 create social containers

## Context


* The goal was to create 2 containers. One for Facebook and one for Twitter. These container will gather all the right data for its children.
* Made for Twitter and Facebook:
  - A container. Gathers all the data.
  - A wrapper with the free component and a slot for premium.
  - A Wordpress SEO component. Includes 'get premium' notification and a form.
  - A Wordpress SEO Premium component. Includes a preview and and a form.
* Made some changes on the JavaScript repo to add logic to components.


## Summary


This PR can be summarized in the following changelog entry:

* [wordpress-seo-premium] Yoast/wordpress-seo-premium#2718
* [javascript] Yoast/javascript#537

## Relevant technical choices:
* Free
  - Made one component for free and one for premium to keep logic seperate.
  - Connected this container to a wrapper that can pass its props on to both the free and premium component.
  - When data changes it's also written to the hidden fields in PHP.
---
* premium
  - A portal connect the Premium with free. In This portal there are 2 Fills to connect with the Slots.
---
* In the javascript repo:
  - ImageSelect has nog a disabled input-field that displays the currently used URL. This only is seen in Free because premium has a preview that displays the image.
  - Made new actions to set the image.
---
* overall: 
  - There still things that need to be done. These are mentioned in https://github.com/Yoast/wordpress-seo/issues/14386. And the will be done in new issues.

## Test instructions

This PR can be tested by following these steps:

1. Checkout all 3 branches out on: 14386 create social containers. ( Free, Premium and javascript ).
1. Merge free into premium with no commit. `git merge free/14386-create-social-containers  --no-commit`
1. Build both free and premium and link the mono-repo.
1. activate Free ( de-activate premium ) and check if the social tabs behave like the old one.
1. De-activate Free and activate premium and check the behaviour again.
1. It should get and display the right data. It should also save that data.

## UI changes

## Documentation

## Quality assurance


Fixes #14386
